src/components/Login.js

```diff
@@ -7,18 +7,50 @@ import {
   InnerForm,
   InnerContent
 } from "./styles";
-import secure from "../image/secure.svg";
-// import {Link } from "react-router-dom"
+import login from "../image/login.svg";
 
-const Login = ({ openSignupModal, showModal, handleClose }) => {
 
+// initializing login values
+const loginInitialValues = {
+  username: "",
+  password: ""
+};
+
+const Login = ({ openSignupModal, showModal,  onLoginSuccess, handleClose }) => {
+
+  const [logininfos, setLoginInfos] = useState(loginInitialValues);
+  const [loading, setLoading] = useState(false);
+
+  //used to redirect to home page
+  // let navigate = useNavigate();
+
+  // fucntion to get user details
+  const setLoginDetails = async (e) => {
+    setLoginInfos({ ...logininfos, [e.target.name]: e.target.value });
+  };
+
+  // function to submit registration form
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
+      const result = await Axios.post(URL+ "/register", userinfos);
+
+      if (result?.status === 200) {
+        setLoading(false);
+        onLoginSuccess()
+      }
+    } catch {
+      console.log("failed to register");
+    }
+    setUserInfos(loginInitialValues);
+  };
 
   return (
     <>
       {showModal ? (
```

Comment:
This `showModal` state should not reside in this component. The component that renders `Login` should handle when it's shown or not. `Login` should only care about closing itself (since it's a modal)